admin/server/usergroups.go

```diff
@@ -493,14 +569,74 @@ func (s *Server) SetProjectMemberUsergroupRole(ctx context.Context, req *adminv1
 		return nil, status.Error(codes.PermissionDenied, "as a non-admin you are not allowed to remove an admin role")
 	}
 
-	err = s.admin.DB.UpdateProjectMemberUsergroup(ctx, usergroup.ID, proj.ID, role.ID)
+	// when setting a full role, remove any resource scoping
+	err = s.admin.DB.UpdateProjectMemberUsergroup(ctx, usergroup.ID, proj.ID, role.ID, nil)
 	if err != nil {
 		return nil, err
 	}
 
 	return &adminv1.SetProjectMemberUsergroupRoleResponse{}, nil
 }
 
+func (s *Server) RemoveProjectMemberUsergroupResources(ctx context.Context, req *adminv1.RemoveProjectMemberUsergroupResourcesRequest) (*adminv1.RemoveProjectMemberUsergroupResourcesResponse, error) {
```

Comment:
I don't think we need add/remove functions for resources. Can it just have `PUT` style behavior and be part of `SetProjectMemberUsergroupRole`? I.e. you pass the full desired resources list on each call.